shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/data/pipeline/ShardingRuleAlteredJobConfigurationPreparer.java

```diff
@@ -75,10 +75,31 @@ public final class ShardingRuleAlteredJobConfigurationPreparer implements RuleAl
     
     @Override
     public void extendJobConfiguration(final YamlRuleAlteredJobConfiguration yamlJobConfig) {
-        Map<String, List<DataNode>> actualDataNodes = getActualDataNodes(new RuleAlteredJobConfigurationSwapper().swapToObject(yamlJobConfig));
+        RuleAlteredJobConfiguration jobConfig = new RuleAlteredJobConfigurationSwapper().swapToObject(yamlJobConfig);
+        Map<String, List<DataNode>> actualDataNodes = getActualDataNodes(jobConfig);
         yamlJobConfig.setJobShardingDataNodes(getJobShardingDataNodes(actualDataNodes));
         yamlJobConfig.setLogicTables(getLogicTables(actualDataNodes.keySet()));
         yamlJobConfig.setTablesFirstDataNodes(getTablesFirstDataNodes(actualDataNodes));
+        ShardingSphereDatabase database = PipelineContext.getContextManager().getMetaDataContexts().getMetaData().getDatabases().get(jobConfig.getDatabaseName());
+        Map<String, String> tableSchema = new HashMap<>(actualDataNodes.size(), 1);
+        for (String logicTable : actualDataNodes.keySet()) {
+            tableSchema.put(logicTable, getLogicTableSchema(database.getSchemas(), new LogicTableName(logicTable)));
+        }
+        yamlJobConfig.setSourceTableSchemaMap(tableSchema);
+        yamlJobConfig.setTargetTableSchemaMap(tableSchema);
```

Comment:
For common fields, it could be put in `RuleAlteredJobWorker.createJobConfig`. `extendJobConfiguration` is used to extend special fields in SPI impl.